UKHomeOffice/nmsw-ui
# NMSW-61 Persist sign in on refresh

# Ticket

NMSW-61

----

## AC

Given I am logged in
When I refresh the page
Then I should remain logged in

Given I am logged in
When I close my browser tab
And I then reopen the nmsw app in a new tab
Then I should NOT be logged in

Given I am logged in
When I click sign out on the nav bar
Then I should be logged out and returned to the landing page

----

## To test

- run app locally
- go to localhost:3000
- click on the start now button
- click on sign in (mocked authentication still)
-> you should see the logged in nav
- refresh your browser
-> you should still be logged in and not redirected to landing page
- close your browser
- open a new browser tab
- go to localhost:3000
-> you should not see the nav (only visible if signed in)
- type localhost:3000/dashboard in the address bar
-> you should remain on the landing page, you cannot access /dashboard if you are not logged in
- click on start now
- sign in
- click on sign out in the nav
-> you should be taken to the landing page
- type localhost:3000/dashboard in the address bar
-> you should remain on the landing page

Added scenario to test
- go to second page form (nav item should be highlighted as second page) 
- fill in form and submit
- click return to homepage button
-> you should be taken to the dashboard
-> the Dashboard item should be highlighted in the nav)

----

## Notes
